servers/rendering_server.cpp

```diff
@@ -1374,6 +1381,30 @@ Error RenderingServer::mesh_create_surface_data_from_arrays(SurfaceData *r_surfa
 	return OK;
 }
 
+int32_t RenderingServer::_get_vertex_to_custom_array_length_factor(uint32_t p_format, int p_array_index) {
+	uint32_t type = (p_format >> (ARRAY_FORMAT_CUSTOM_BASE + ARRAY_FORMAT_CUSTOM_BITS * (p_array_index - RS::ARRAY_CUSTOM0))) & ARRAY_FORMAT_CUSTOM_MASK;
+	switch (type) {
+		case ARRAY_CUSTOM_RGBA8_UNORM:
+		case ARRAY_CUSTOM_RGBA8_SNORM:
+		case ARRAY_CUSTOM_RG_HALF: {
+			return 4;
+		} break;
+		case ARRAY_CUSTOM_RGBA_HALF: {
+			return 8;
+		} break;
+		case ARRAY_CUSTOM_R_FLOAT:
+		case ARRAY_CUSTOM_RG_FLOAT:
+		case ARRAY_CUSTOM_RGB_FLOAT:
+		case ARRAY_CUSTOM_RGBA_FLOAT: {
+			int32_t s = type - ARRAY_CUSTOM_R_FLOAT + 1;
+
+			return s;
+		} break;
+		default: {
+		}
```

Comment:
You need to return something here or throw an error, like so:
```suggestion
			ERR_FAIL_V_MSG(0, "Invalid texture type.");
		}
```
With some appropriate message

Reply:
Thanks. I decided to call it a “custom format type” instead, as I believe that describes it better.